extensions/amp-story/1.0/amp-story-request-service.js

```diff
@@ -50,26 +55,42 @@ export class AmpStoryRequestService {
    * @private
    */
   loadBookendConfigImpl_() {
-    return this.loadJsonFromAttribute_(BOOKEND_CONFIG_ATTRIBUTE_NAME);
+    const bookendEl = childElementByTag(this.storyElement_,
+        'amp-story-bookend');
+    if (!bookendEl) {
+      return Promise.resolve(null);
+    }
+
+    if (bookendEl.hasAttribute(BOOKEND_CONFIG_ATTRIBUTE_NAME)) {
+      return this.loadJsonFromAttribute_(
+          BOOKEND_CONFIG_ATTRIBUTE_NAME, bookendEl);
+    }
+
+    // Fallback. Check for an inline json config.
+    let config = null;
+    try {
+      config = getChildJsonConfig(bookendEl);
+    } catch (err) {}
+
+    return Promise.resolve(config);
   }
 
   /**
    * @param {string} attributeName
+   * @param {!Element} bookendEl
    * @return {(!Promise<!JsonObject>|!Promise<null>)}
    * @private
    */
-  loadJsonFromAttribute_(attributeName) {
-    const bookendEl = childElementByTag(this.storyElement_,
-        'amp-story-bookend');
-
-    if (!bookendEl || !bookendEl.hasAttribute(attributeName)) {
-      return Promise.resolve(null);
-    }
-
+  loadJsonFromAttribute_(attributeName, bookendEl) {
     const rawUrl = bookendEl.getAttribute(attributeName);
     const opts = {};
     opts.requireAmpResponseSourceOrigin = false;
 
+    if (!isProtocolValid(rawUrl)) {
+      user().error(TAG, 'Invalid bookend config url.');
```

Comment:
Nit: please make the message more generic, as this service should be reusable : )